static/app/components/events/interfaces/spans/spanTreeModel.tsx

```diff
@@ -264,6 +294,49 @@ class SpanTreeModel {
 
     return [wrappedSpan, ...descendants];
   };
+
+  toggleEmbeddedChildren = (props: {orgSlug: string; eventSlug: string}) => {
+    this.showEmbeddedChildren = !this.showEmbeddedChildren;
+
+    if (this.showEmbeddedChildren && this.embeddedChildren.length === 0) {
+      return this.fetchEmbeddedTransactions(props);
+    }
+
+    return Promise.resolve(undefined);
+  };
+
+  fetchEmbeddedTransactions({orgSlug, eventSlug}: {orgSlug: string; eventSlug: string}) {
+    const url = `/organizations/${orgSlug}/events/${eventSlug}/`;
+
+    this.loadingEmbeddedChildren = true;
+
+    return this.api
+      .requestPromise(url, {
+        method: 'GET',
+        query: {},
+      })
+      .then(
+        action('fetchEmbeddedTransactionsSuccess', (event: EventTransaction) => {
+          if (!event) {
+            return;
+          }
+
+          const parsedTrace = parseTrace(event);
+          const rootSpan = generateRootSpan(parsedTrace);
+          const parsedRootSpan = new SpanTreeModel(
+            rootSpan,
+            parsedTrace.childSpans,
+            this.api,
+            false
+          );
+
+          this.embeddedChildren = [parsedRootSpan];
+        })
+      )
```

Comment:
This can be a followup but should this handle the `.catch` case and display an error or something?

Reply:
There definitely should be a `.catch()` here. I refactored to roll up the loading and error states into `this.fetchEmbeddedChildrenState`.